### PR TITLE
Fixes #4 Hover implementation

### LINKED ISSUE #4: POC: Set up framework for hover
**Proof of Concept**
1. Create a sample hover object
- create a new `org.eclipse.jdt.ui.javaEditorTextHovers` in the `plugin.xml` (see org.eclipse.jdt.ui.javaEditorTextHovers)
ie. 
```
<extension
         id="jakartajava-hover-provider"
         point="org.eclipse.jdt.ui.javaEditorTextHovers">
      <hover
            activate="true"
            class="org.jakarta.lsp4e.JakartaJavaHoverProvider"
            id="org.jakarta.lsp4e.hoverprovider">
      </hover>
   </extension>
```
- create a new `JakartaJavaHoverProvider` class, see https://github.com/spring-projects/sts4/blob/6719e0afb393376dc2323bb5ea89de3d309dc98d/eclipse-language-servers/org.springframework.tooling.boot.ls/src/org/springframework/tooling/boot/ls/jdt/SpringBootJavaHoverProvider.java for reference
- implement a hover from [JakartaTextDocumentService](https://github.com/MicroShed/jakarta-ls/blob/master/jakarta.ls/src/main/java/io/microshed/jakartals/JakartaTextDocumentService.java) via the `getJavaHover()` method

### PR DESCRIPTION
**Pull request by PENGYUXIONG**
- Inserted hover dependency inside xml
- added hover function inside JakartaTextDocumentService.java for
service end
- modified the getJavaHover function inside
JakartaLanguageClientAPI.java (server side) and JakartaLanguageClient.java (client side) to return customized testing hover model
- added new hover JakartaJavaHoverProvider.java at client side to specify
the hovering style

### CODE REVIEW

File: jakarta-eclipse/src/org/jakarta/lsp4e/JakartaJavaHoverProvider.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+public class JakartaJavaHoverProvider implements IJavaEditorTextHover {
+	private LSPTextHover lspTextHover;
+	
+	public JakartaJavaHoverProvider() {
+		super();
+		lspTextHover = new LSPTextHover();
+	}
+
+	@Override
+	public String getHoverInfo(ITextViewer arg0, IRegion arg1) {
```
Comment (rzgry) [inline jakarta-eclipse/src/org/jakarta/lsp4e/JakartaJavaHoverProvider.java:19]: public String getHoverInfo(ITextViewer textViewer, IRegion region) {

Also remove the TODO messages

File: jakarta-eclipse/src/org/jakarta/lsp4e/JakartaJavaHoverProvider.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		// TODO Auto-generated method stub
+		return this.lspTextHover.getHoverInfo(arg0, arg1);
+	}
+
+	@Override
+	public IRegion getHoverRegion(ITextViewer arg0, int arg1) {
```
Comment (rzgry) [inline jakarta-eclipse/src/org/jakarta/lsp4e/JakartaJavaHoverProvider.java:25]: public IRegion getHoverRegion(ITextViewer textViewer, int offset) {

File: jakarta-eclipse/src/org/jakarta/lsp4e/JakartaJavaHoverProvider.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		// TODO Auto-generated method stub
+		return this.lspTextHover.getHoverRegion(arg0, arg1);
+	}
+
+	@Override
+	public void setEditor(IEditorPart arg0) {
```
Comment (rzgry) [inline jakarta-eclipse/src/org/jakarta/lsp4e/JakartaJavaHoverProvider.java:31]: public void setEditor(IEditorPart editor)

File: jakarta.ls/src/main/java/io/microshed/jakartals/JakartaTextDocumentService.java [hunk 4]
```diff
@@ -93,6 +93,21 @@ private void triggerValidationForAll() {
...
+		LOGGER.info("Getting the hover going at server");
+		return jakartaLanguageServer.getLanguageClient().getJavaHover(params)
+				.thenApply(hover -> {
+					// if hover val is null
+					if (hover == null){
+						return null;
+					}
+					return hover;
+				});
+	}
```
Comment (rzgry) [inline jakarta.ls/src/main/java/io/microshed/jakartals/JakartaTextDocumentService.java:110]: This code is unnecessary. It always just returns the value of `hover`.
```
.thenApply(hover -> {
					// if hover val is null
					if (hover == null){
						return null;
					}
					return hover;
				});
```

Can just be `return jakartaLanguageServer.getLanguageClient().getJavaHover(params)`

File: jakarta.ls/src/main/java/io/microshed/jakartals/JakartaTextDocumentService.java [hunk 5]
```diff
@@ -93,6 +93,21 @@ private void triggerValidationForAll() {
 		List<String> allDocs = documents.all().stream().map(doc -> doc.getUri()).collect(Collectors.toList());
 		triggerValidationFor(allDocs);
 	}
+	
+	
+	@Override
+	public CompletableFuture<Hover> hover(HoverParams params) {
+		// TODO give hover message to the server
+		LOGGER.info("Getting the hover going at server");
```
Comment (rzgry) [inline jakarta.ls/src/main/java/io/microshed/jakartals/JakartaTextDocumentService.java:101]: Remove this log message or change it to something like `received textDocument/hover request`

**Comment by rzgry** [comment]
Also you need to fix ur DCO check. You can run the commands from here to fix it: https://github.com/MicroShed/jakarta-ls/runs/1184059430

**Review by kathrynkodama — APPROVED** [review]

**Pull request merged**